src/wwwroot/app/services/ThesaurusService.test.js

```diff
@@ -0,0 +1,166 @@
+      },
+      reject: function () {
+         return this;
+      }
+   };
+   let mockHttp = {
+      get: jasmine.createSpy().and.returnValue(promiseMock),
+      post: jasmine.createSpy().and.returnValue(promiseMock),
+      put: jasmine.createSpy().and.returnValue(promiseMock),
+      remove: jasmine.createSpy().and.returnValue(promiseMock)
+   };
+   let mockTranslate = {
+      instant: jasmine.createSpy()
+   };
+
+   function _sendToCacheThesaurus(thesaurusName) {
+      spyOn(promiseMock, 'then');
+      service.getThesaurusTopics(thesaurusName);
+      let callback = promiseMock.then.calls.first().args[0];
+      let response = {};
+      response[thesaurusName] = [];
+      callback(response);
+      promiseMock.then.calls.reset();  
+   }
+
+   beforeEach(() => {
+      angular.module('test', []).service('ThesaurusService', thesaurusService);
+      angular.mock.module('test');
+      angular.mock.module($provide => {
+         $provide.value('HttpService', mockHttp);
+         $provide.value('$translate', mockTranslate);
+         $provide.value('$q', promiseMock);
+      });
+   });
+
+   beforeEach(inject(ThesaurusService => {
+      service = ThesaurusService;
+   }));
+
+   it('getThesaurusTopics not to be undefined or null', function() {
+      expect(service.getThesaurusTopics).not.toBeUndefined();
+      expect(service.getThesaurusTopics).not.toBeNull();
+   });
+
+   it('getThesaurusTopic not to be undefined or null', function() {
+      expect(service.getThesaurusTopic).not.toBeUndefined();
+      expect(service.getThesaurusTopic).not.toBeNull();
+   });
+
+   it('getThesaurusTopicsGroup not to be undefined or null', function() {
+      expect(service.getThesaurusTopicsGroup).not.toBeUndefined();
+      expect(service.getThesaurusTopicsGroup).not.toBeNull();
+   });
+
+   it('saveThesaurusTopic not to be undefined or null', function() {
+      expect(service.saveThesaurusTopic).not.toBeUndefined();
+      expect(service.saveThesaurusTopic).not.toBeNull();
+   });
+
+   it('saveThesaurusTopics not to be undefined or null', function() {
+      expect(service.saveThesaurusTopics).not.toBeUndefined();
+      expect(service.saveThesaurusTopics).not.toBeNull();
+   });
+
+   it('deleteThesaurusTopic not to be undefined or null', function() {
+      expect(service.deleteThesaurusTopic).not.toBeUndefined();
+      expect(service.deleteThesaurusTopic).not.toBeNull();
+   });
+
+   it('getThesaurusTopics call test for simple thesaurus', function() {
+      service.getThesaurusTopics('socialnetworks');
+      expect(mockHttp.get).toHaveBeenCalledWith('socialnetworks');
+   });
+
+   it('getThesaurusTopics call test for complex thesaurus', function() {
+      service.getThesaurusTopics('locations')
+      expect(mockHttp.get).toHaveBeenCalledWith('locations');
+      expect(mockHttp.get).toHaveBeenCalledWith('countries');
+   });
+
+   it('test cache: http calls once', function() {
+      mockHttp.get.calls.reset();
```

Comment:
I think you need to use it in `beforeEach` block. this will produce more independent test